compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -884,27 +883,8 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         binder: NodeId,
         generic_params: &[GenericParam],
     ) -> &'hir [hir::GenericParam<'hir>] {
-        let mut generic_params: Vec<_> = generic_params
```

Comment:
Moving this into `lower_generic_param_kind` since `lower_lifetime_binder` doesn't get called on HRTBs.